common/lib/xmodule/xmodule/html_module.py

```diff
@@ -75,7 +75,17 @@ class HtmlModuleMixin(HtmlFields, XModule):
 
     def get_html(self):
         if self.system.substitute_keywords_with_data:
-            return self.system.substitute_keywords_with_data(self.data)
+            course = self.descriptor.runtime.modulestore.get_course(self.course_id)
+            user = self.system.get_real_user(self.system.anonymous_student_id)
+            context = {
+                'user_id': user.id,
```

Comment:
I've got a few tests breaking here with:

```
AttributeError: 'NoneType' object has no attribute 'id'
```

It looks like the call to `get_real_user` is returning `None`, because `self.system.anonymous_student_id` is also `None`.